ui/app/models/cluster.js

```diff
@@ -49,46 +49,9 @@ export default DS.Model.extend({
 
   anyReplicationEnabled: or('{dr,performance}.replicationEnabled'),
 
-  stateDisplay(state) {
-    if (!state) {
-      return null;
-    }
-    const defaultDisp = 'Synced';
-    const displays = {
-      'stream-wals': 'Streaming',
-      'merkle-diff': 'Determining sync status',
-      'merkle-sync': 'Syncing',
-    };
-
-    return displays[state] || defaultDisp;
-  },
-
-  drStateDisplay: computed('dr.state', function() {
-    return this.stateDisplay(this.get('dr.state'));
-  }),
-
-  performanceStateDisplay: computed('performance.state', function() {
-    return this.stateDisplay(this.get('performance.state'));
-  }),
-
-  stateGlyph(state) {
-    const glyph = 'check-circle-outline';
-
-    const glyphs = {
-      'stream-wals': 'check-circle-outline',
-      'merkle-diff': 'android-sync',
-      'merkle-sync': 'android-sync',
-    };
-
-    return glyphs[state] || glyph;
-  },
-
-  drStateGlyph: computed('dr.state', function() {
-    return this.stateGlyph(this.get('dr.state'));
-  }),
-
-  performanceStateGlyph: computed('performance.state', function() {
-    return this.stateGlyph(this.get('performance.state'));
+  modeState: computed('dr.{mode,state}', 'performance.{mode,state}', 'replicationMode', function() {
```

Comment:
Since only `replicationMode` is accessed for the value, I don't think `dr{...}` and `performance{...}` are necessary. Also I've never seen that `thing.{foo,bar}` pattern before 🤯

Reply:
glad you brought this up! i think we still need all of these properties here because `modeState` is dependent on all of their values. that is, when the `dr.state` or `performance.state` changes, we want to update our cache so that the next time `modeState` is accessed in the template, the new computed value is shown. i know it's wonky since we're only using `replicationMode` on line 54, but behind the scenes we want the `drOrPerf.mode` to affect this property. i found this blurb on the ember guides that talks about this process: 

>The first time you access the fullName property, the function will be called and the results will be cached. Subsequent access of fullName will read from the cache without calling the function. Changing any of the dependent properties causes the cache to invalidate, so that the computed function runs again on the next access.

that being said i am not 100% positive about this, so if you think i've misunderstood definitely let me know!

regarding the `thing.{stuff, yay}` syntax, my linter showed me that! there's more about using object properties as dependent keys and using that syntax here: 
https://guides.emberjs.com/v3.8.0/object-model/computed-properties-and-aggregate-data/